common/src/main/java/org/jetbrains/annotations/CheckReturnValue.java

```diff
@@ -0,0 +1,30 @@
+package org.jetbrains.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Collection;
+
+/**
+ * Specifies that the return value of the method should be used.
+ * <p>
+ * Typical use cases are:
+ * <ul>
+ *   <li>pure or side-effect-free methods
+ *   <li>methods that use the return value for reporting success or error
+ *   <li>methods that return an unfinished computation,
+ *       such as {@link Collection#stream()} or builders
```

Comment:
This is basically the same as "pure or side-effect-free" methods, so I don't see the difference. I would actually concentrate on side-effect methods, like

> Specifies that the return value of the impure method should be used. For pure method (annotated with `@Contract(pure = true)`), it's implied that the resulting value is important, so this annotation is unnecessary. However, sometimes it's important for impure method as well. For example, `InputStream#read(byte[])` returns the number of bytes actually stored into byte array. Without checking the return value, it's impossible to say how many bytes were actually read.

WDYT?

Reply:
Sounds good, thanks for the example.